feature/preview/src/main/java/com/google/jetpackcamera/feature/preview/PreviewScreen.kt

```diff
@@ -107,14 +102,15 @@ fun PreviewScreen(
         zoomHandler.postDelayed({ zoomScaleShow = false }, 3000)
     })
 
-    LaunchedEffect(lifecycleOwner) {
-        val surfaceProvider = deferredSurfaceProvider.await()
-        lifecycleOwner.repeatOnLifecycle(Lifecycle.State.STARTED) {
-            viewModel.runCamera(surfaceProvider)
-            try {
-                awaitCancellation()
-            } finally {
-                viewModel.stopCamera()
+    LaunchedEffect(lifecycleOwner, surfaceProvider) {
+        if (surfaceProvider != null) {
+            lifecycleOwner.repeatOnLifecycle(Lifecycle.State.STARTED) {
+                viewModel.runCamera(surfaceProvider)
+                try {
+                    awaitCancellation()
+                } finally {
+                    viewModel.stopCamera()
```

Comment:
If I understand this snippet correctly, this means you'd only stop the camera in the on-destroy lifecycle event. Is that okay, or would it need to be stopped during the on-stop lifecycle event?

Reply:
We will stop the camera in onStop, `repeatOnLifecycle(Lifecycle.State.STARTED)` launches a new coroutine in the block that will be cancelled when lifecycle state is in STOPPED.

Based this off of: https://developer.android.com/topic/libraries/architecture/coroutines#restart

Reply:
Ah yes, you're right. I got confused when reading the documentation on `withStarted` which mentions onDestroy instead of onStop:

```
Run block with this LifecycleOwner's Lifecycle in a Lifecycle.State of at least Lifecycle.State.STARTED and resume with the result. Throws the CancellationException if the lifecycle has reached Lifecycle.State.DESTROYED by the time of the call or before block is able to run.
```